src/main/java/projectbookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package projectbookstore.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import projectbookstore.dto.UserRegistrationRequestDto;
+import projectbookstore.dto.UserResponseDto;
+import projectbookstore.exception.RegistrationException;
+import projectbookstore.mapper.UserMapper;
+import projectbookstore.model.User;
+import projectbookstore.repository.UserRepository;
+import projectbookstore.service.UserService;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto request)
+            throws RegistrationException {
+        if (userRepository.findByEmail(request.getEmail()).isPresent()) {
```

Comment:
we can use exists here because we do not need to fetch the actual data of that user, just the fact that it is present or no
```suggestion
        if (userRepository.existsByEmail(request.getEmail())) {
```